### PR TITLE
Commit processing: check for expiration date; check for timeout

### PR DESCRIPTION
**Pull request by alex-moc**

### CODE REVIEW

File: FitpaySDK/PaymentDevice/PaymentDevice.swift [hunk 1]
```diff
@@ -282,8 +282,19 @@
                 
                 self?.apduResponseHandler = completion
                 log.verbose("APDU_DATA: Calling device interface to execute APDU's.")
-                self?.deviceInterface.executeAPDUCommand(apduCommand)
                 
+                var isCompleteExecute = false
+                DispatchQueue.global().asyncAfter(deadline: .now() + .seconds(20), execute: {
```
Comment (ssteveli) [inline FitpaySDK/PaymentDevice/PaymentDevice.swift]: This value should be configurable with a higher default, we use 30 seconds as the default for Android.

File: FitpaySDK/PaymentDevice/PaymentDevice.swift [hunk 2]
```diff
@@ -282,8 +282,19 @@
...
+                    if !isCompleteExecute {
+                        self?.apduResponseHandler = nil
+                        log.verbose("APDU_DATA: Received timeout during execute APDU's.")
```
Comment (ssteveli) [inline FitpaySDK/PaymentDevice/PaymentDevice.swift]: So this is applied only to "APDU" handling?   If so, this should be applied for all commits that an OEM is listening and responding too?

Comment (alex-moc) [reply FitpaySDK/PaymentDevice/PaymentDevice.swift]: Fixed in new commit

File: FitpaySDK/PaymentDevice/PaymentDevice.swift [hunk 3]
```diff
@@ -284,10 +287,10 @@
                 log.verbose("APDU_DATA: Calling device interface to execute APDU's.")
                 
                 var isCompleteExecute = false
-                DispatchQueue.global().asyncAfter(deadline: .now() + .seconds(20), execute: {
+                DispatchQueue.global().asyncAfter(deadline: .now() + .seconds(30), execute: {
```
Comment (ssteveli) [inline FitpaySDK/PaymentDevice/PaymentDevice.swift]: This 30 seconds can't be hard coded like this, it needs to be configurable.

File: FitpaySDK/PaymentDevice/PaymentDevice.swift [hunk 4]
```diff
@@ -305,7 +308,17 @@
     
     internal func processNonAPDUCommit(commit: Commit, completion: @escaping (_ state: NonAPDUCommitState?, _ error: NSError?) -> Void) {
         if let processNonAPDUCommit = self.deviceInterface.processNonAPDUCommit {
+
+            var isCompleteProcessing = false
+            DispatchQueue.global().asyncAfter(deadline: .now() + .seconds(30), execute: {
```
Comment (ssteveli) [inline FitpaySDK/PaymentDevice/PaymentDevice.swift]: This hard coded 30 seconds needs to be configurable, 30 seconds is only the default.

**Pull request merged**